internal/handlers/handler.go

```diff
@@ -0,0 +1,71 @@
+package handler
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"io"
+	"net/http"
+	"net/url"
+	"sync"
+
+	"github.com/go-chi/chi/v5"
+)
+
+var str = map[string]string{}
+var lock = sync.RWMutex{}
+
+// Handler — обработчик запроса.
+func Handler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodPost:
+		su, err := io.ReadAll(r.Body)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		defer r.Body.Close()
+
+		u, err := url.ParseRequestURI(string(su))
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		shu := GetMD5Hash(u)
+
+		lock.RLock()
```

Comment:
```suggestion
		lock.Lock()
```
Здесь происходит запись в структуру, поэтому лочить надо полноценно. RLock используется для ускорения совместного чтения, так как два им более одновременных RLock друг друга не задерживают.

Reply:
https://medium.com/a-journey-with-go/go-concurrency-access-with-maps-part-iii-8c0a0e4eb27e
https://gobyexample.com/mutexes

Reply:
поняла, спасибо за пояснения!